numpy/core/tests/test_regression.py

```diff
@@ -2115,7 +2115,9 @@ def __array__(self,*args,**kwargs):
         for f in [op.lt, op.le, op.gt, op.ge]:
             if sys.version_info[0] >= 3:
                 assert_raises(TypeError, f, lhs, rhs)
-            else:
+            elif not sys.py3kwarning:
+                # With -3 switch in python 2, DeprecationWarning is raised
+                # which we are not interested in
```

Comment:
I could not add a warning context here as you commented last time. So the idea was to skip this test under -3 switch.